module-gtnhmixins/src/main/java/com/gtnewhorizon/gtnhmixins/mixins/LateMixinOrchestrationMixin.java

```diff
@@ -91,4 +100,29 @@ private void beforeConstructing(Object[] eventData, CallbackInfo ci) throws Thro
         Reflection.invokeSelectConfigs(transformer, env);
         Reflection.invokePrepareConfigs(transformer, env);
     }
+
+    @SuppressWarnings("unchecked")
+    private static Set<String> getLiteLoaderMods() {
+        Set<String> mods = new HashSet<>();
+        try {
+            Class<?> LiteLoaderTweaker = Class.forName("com.mumfrey.liteloader.launch.LiteLoaderTweaker");
+            Class<?> LoadableModClassPath = Class.forName("com.mumfrey.liteloader.core.api.LoadableModClassPath");
+            Object instance = FieldUtils.readDeclaredStaticField(LiteLoaderTweaker, "instance", true);
+            Object bootstrap = FieldUtils.readDeclaredField(instance, "bootstrap", true);
+            Object enumerator = FieldUtils.readDeclaredField(bootstrap, "enumerator", true);
+            Map<String, ?> enabledContainers = (Map<String, ?>) FieldUtils.readDeclaredField(enumerator, "enabledContainers", true);
+            GTNHMixins.LOGGER.info("LiteLoader present, adding its mods to the list.");
+            for(Entry<String, ?> e : enabledContainers.entrySet()) {
+                // Classpath mods include e.g. all libraries (lwjgl, guava, etc.) - we don't want those 
+                if(!LoadableModClassPath.isInstance(e.getValue())) {
```

Comment:
I'm not too familiar with LiteLoader, but this doesn't seem correct. In Forge's case, there can be valid mods on the classpath if they are added as runtime dependencies in the buildscript, and I assume the same is true for LiteLoader. Those mods would get ignored by this code.

`((LoadableModFile)e.getValue()).hasValidMetaData()` seems more like what you want, it returns true only for jars that are valid LiteLoader mods.